packages/legend-query-builder/src/components/explorer/QueryBuilderPropertySearchPanel.tsx

```diff
@@ -283,31 +290,51 @@ export const QueryBuilderPropertySearchPanel = observer(
     const explorerState = queryBuilderState.explorerState;
     const propertySearchPanelState = explorerState.propertySearchState;
     const searchInputRef = useRef<HTMLInputElement>(null);
-    const changePropertyName: React.ChangeEventHandler<HTMLInputElement> = (
-      event,
-    ) => {
+
+    const handleEnter = (): void => searchInputRef.current?.focus();
+
+    const debouncedSearchProperty = useMemo(
+      () => debounce(() => propertySearchPanelState.search(), 100),
+      [propertySearchPanelState],
+    );
+    const onSearchPropertyTextChange: React.ChangeEventHandler<
+      HTMLInputElement
+    > = (event) => {
       propertySearchPanelState.setSearchText(event.target.value);
-      propertySearchPanelState.refreshPropertyState();
-      if (propertySearchPanelState.searchText.length >= 3) {
-        propertySearchPanelState.fetchMappedPropertyNodes(
-          propertySearchPanelState.searchText,
-        );
-      }
+      debouncedSearchProperty();
     };
-    const clearPropertyName = (): void => {
+
+    const clearSearchPropertyName = (): void => {
       propertySearchPanelState.setSearchText('');
-      propertySearchPanelState.refreshPropertyState();
+      propertySearchPanelState.resetPropertyState();
     };
+
     const handleClose = (): void => {
-      clearPropertyName();
+      clearSearchPropertyName();
       propertySearchPanelState.setIsSearchPanelOpen(false);
     };
+
+    const getDocumentation = (): void => {
+      queryBuilderState.applicationStore.assistantService.setIsOpen(true);
+
+      queryBuilderState.applicationStore.assistantService.getDocumentationFromKey(
+        'question.how-to-use-search-bar',
```

Comment:
--> this should be part of the `TextSearchAdvancedConfig` component and the key should be taken from the enum